nomic/atlas.py

```diff
@@ -12,10 +12,144 @@
 from loguru import logger
 from tqdm import tqdm
 
-from .project import AtlasProject
+from .dataset import AtlasDataset
 from .settings import *
 from .utils import b64int, get_random_name, arrow_iterator
 
+from .data_inference import NomicProjectHyperparameters, NomicTopicHyperparameters, NomicDuplicatesHyperparameters
+
+
+def map_data(
+    data: List[Dict] = None,
+    embeddings: np.array = None,
+    name: str = None,
+    description: str = "",
+    id_field: str = None,
+    is_public: bool = True,
+    indexed_field: str = None,
+    projection: Optional[Dict] = None,
+    topic_model: Optional[Dict] = None,
+    duplicate_detection: Optional[Dict] = None
+
+) -> AtlasDataset:
+    """
+
+    Args:
+        data: An [N,] element list of dictionaries containing metadata for each datapoint.
+        embeddings: An [N,d] numpy array containing the N embeddings to add.
+        name: The name of your dataset
+        description: The description of your dataset
+        id_field: Specify your data unique id field. This field can be up 36 characters in length. If not specified, one will be created for you named `id_`.
+        is_public: Should the dataset be accessible outside your Nomic Atlas organization.
+        projection: Hyperparameters to adjust Nomic Project - the dimensionality algorithm organizing your dataset.
+        topic_model: Hyperparameters to adjust Nomic Topic - the topic model organizing your dataset.
+        duplicate_detection: Hyperparameters to adjust Nomic Duplicates - the duplicate detection algorithm.
+    :return:
+    """
+    if embeddings is not None:
+        modality = 'embedding'
+        assert isinstance(embeddings, np.ndarray), 'You must pass in a numpy array'
+        if embeddings.size == 0:
+            raise Exception("Your embeddings cannot be empty")
+
+    if indexed_field is not None:
+        modality = 'text'
+
+    if id_field is None:
+        id_field = ATLAS_DEFAULT_ID_FIELD
+
+    project_name = get_random_name()
+    if description is None:
+        description = 'A description for your map.'
```

Comment:
Description cannot be `None` according the type signature. I think we should remove this line anyway, an empty string or NULL or better than this placeholder.